### DISCUSSION TITLE
Actual exposed JSON model is not complete compared to DEBUG view

### DISCUSSION THREAD

**Discussion by SjoerdV** [microsoft-search/pnp-modern-search#1965] [Q&A]
**Version used**
Ex: 4.4.1

**Describe the bug**
When using the DEBUG view of the results webpart this is the modal that is shown
[Screen 1]

When using the following handlebars ```{{JSONstringify @root}}``` the following modal is available. Quite a lot of nodes/keys are missing here and can not be traversed
[Screen 2]

**To Reproduce**
1. add a Results webpart with a fixed query template, for instance using the ```ContentTypeId:0x0100....``` query
2. Limit the number of results to 1 (to prevent overload of info)
3. Choose the 'Details list' layout
4. Use 'Manage Columns' and add/change a column to include this handlebar template ```{{JSONstringify @root}}```, doesn't matter were just have it parse out all properties.
5. Copy the JSON properties that are displayed on screen to notepad or IDE and format the results (should depict [Screen 1])
6. Now switch the layout to DEBUG view and notice a lot more properties available (should depict [Screen 2])

**Expected behavior**
I would expect to be able to use the '@root.properties.dataSourceProperties.queryTemplate' (as an example) inside my handlebars constructs, except that the 'properties' node (amongst others) is not exposed. Why can we only use 'slots', 'theme' and 'context'?

**Screenshots**
[Screen 1]
![image](https://user-images.githubusercontent.com/13275320/160023720-1c37be02-3bee-4b9c-b395-7a5067a462cc.png)
[Screen 2]
![image](https://user-images.githubusercontent.com/13275320/160023681-55c381f1-2ce0-4d61-a45f-3caab4f23595.png)

**Desktop (please complete the following information):**
 - Firefox, Edge Chromium

**Additional context**

**Comment by wobba** [comment]
Like this you mean? Which works fine in the latest version at least.
![image](https://user-images.githubusercontent.com/51104/160236495-bfc15f4b-380d-4f14-ad78-95503874107e.png)

**Reply by SjoerdV** [reply]
Ah ok, I didn't test this in the overall template. 
That exact handlebars part seems not to work when used in the 'Details' layout 'Manage Columns' specific config. Is that correct?

**Reply by wobba** [reply]
That is likely correct. Can you update your question with your exact setup?

**Reply by SjoerdV** [reply]
Well, I actually don't have a setup, because what I was thinking I could do with a column configuration didn't work. 

The bottom line is as follows: 
- I add anything from the current page properties like ```OR xxx:{Page.ID}``` to the search query in the mentioned Results webpart (xxx is not defined, but that does not matter)
- then use the ```regex``` handlebar helper to extract that string replaced ```{Page.ID}``` bit from the query and use that in the Column value handlebars, maybe as part of a url link.

So much possibilities would open up that would not require going into the search schema (which surely is the tedious bit, right)

**I would like to know though, why there should be a difference in exposing the json model to the overall template but limit this model to the column templates.**

Thanks for this great solution btw. ```'vanilla' SharePoint -> SharePoint + PnP Search``` is like ```Lego -> Technic Lego```

**Reply by wobba** [reply]
The difference is where in the code we parse things and how. Not saying they shouldn't/couldn't be the same. Maybe @FranckyC has some input.

**Reply by FranckyC** [reply]
At the time, there was no need to expose all the context to the details or cards to keep a small context with only relevant properties. However, after reviewing the code, it shouldn't be hard to expose everything as we already have the context here (coming from the `data-context` attribute):

![image](https://user-images.githubusercontent.com/5659122/160683558-8a1d03f6-060b-4d30-a1f4-7392662172ef.png)

**Reply by FranckyC** [reply]
@SjoerdV here you go https://github.com/microsoft-search/pnp-modern-search/pull/1979

**Reply by SjoerdV** [reply]
Great show! Thanks @wobba and @FranckyC.
